src/btree/bt_compact.c

```diff
@@ -187,6 +209,14 @@ __wt_compact(WT_SESSION_IMPL *session)
 
 		session->compact_state = WT_COMPACT_SUCCESS;
 		WT_STAT_DATA_INCR(session, btree_compact_rewrite);
+
+		if (WT_VERBOSE_ISSET(session, WT_VERB_COMPACT_PROGRESS)) {
+			now = __wt_clock(session);
+			time_diff = WT_CLOCKDIFF_SEC(now, last);
+			if (time_diff >= 1)
+				__compact_progress(session, time_diff);
+			last = now;
```

Comment:
Shouldn't this only be updated when a message is actually output?

I think the timer variable should be moved to the connection - otherwise there might be 1000 calls to this function that cumulatively take a long time, but none of them are individually long enough to report a message.

I'd also prefer for all of this logic to be within the `__compact_progress` function - similar to the checkpoint progress message.

Reply:
@agorrod I have added the timer into the `compact` state in the `session` structure and moved the logic within the `__compact_progress` function. Take a look and LGTM if looks OK.